app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -107,34 +119,62 @@ class TopicListAdapter(
     return itemList.size
   }
 
-  private class WelcomeViewHolder(
-    val binding: WelcomeBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
+  private class WelcomeViewHolder(val binding: WelcomeBinding) : RecyclerView.ViewHolder(binding.root) {
     internal fun bind(userAppHistoryViewModel: UserAppHistoryViewModel) {
       binding.viewModel = userAppHistoryViewModel
     }
   }
 
-  private class PromotedStoryViewHolder(
-    val binding: PromotedStoryCardBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
-    internal fun bind(promotedStoryViewModel: PromotedStoryViewModel) {
-      binding.viewModel = promotedStoryViewModel
+  inner class PromotedStoryListViewHolder(val binding: PromotedStoryListBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(
+      activity: AppCompatActivity,
+      promotedStoryListViewModel: PromotedStoryListViewModel,
+      promotedStoryList: MutableList<PromotedStoryViewModel>
+    ) {
+      binding.viewModel = promotedStoryListViewModel
+      val promotedStoryAdapter = PromotedStoryListAdapter(promotedStoryList)
+      val horizontalLayoutManager = LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
```

Comment:
Nit: please add inline comment argument for the false literal.

Reply:
Done.